components/config-mgmt-service/integration_test/node_export_test.go

```diff
@@ -689,3 +689,84 @@ func TestNodeExportLoopBug(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeExportPaging(t *testing.T) {
+	esClient := elastic.New(elasticsearchUrl)
+	c := config.New(esClient)
+	server := grpcserver.NewCfgMgmtServer(c)
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	api.RegisterCfgMgmtServer(s, server)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+
+	dialer := func(string, time.Duration) (net.Conn, error) { return lis.Dial() }
+
+	conn, err := grpc.DialContext(context.Background(), "bufnet", grpc.WithDialer(dialer), grpc.WithInsecure())
+	defer conn.Close()
+	require.NoError(t, err)
+
+	client := api.NewCfgMgmtClient(conn)
+
+	names := []string{
+		"Taco Bell",
+		"taco bell",
+		"TACO BELL",
+		"tACO bELL",
+		"TACO bell",
+	}
+
+	// page size is hardcoded to 100, so to test the pagination code we need more than that.
+	numberOfNodes := 150
```

Comment:
The number 150 was not chosen super specifically, but with 5 possible node names and 150 nodes I was able to see this test fail reliably and I couldn't always get it to fail with fewer nodes.